### ISSUE TITLE
access token, advanced caching, sorting

### ISSUE THREAD

**Issue by reluem** [fritzmg/contao-facebook-gallery#5]
Hi, tried your module today & worked fine with some code adjustments. 👍 

Bug:
- the facebook access code was not properly generated. On entering the details, the error log showed the following errors: 
> PHP Warning: file_get_contents(https://graph.facebook.com/xxx?fields=id,name&{"access_token":"xxx|xxx","token_type":"bearer"}): failed to open stream: HTTP request failed! HTTP/1.1 403 Forbidden
I think the access token is not properly pulled from the getAccessToken(), because the token type is also included and the token is in curly braces.

-> I manually changed the getAccessToken() to the token I received from Facebook for a (quick&dirty) test, which worked fine!

Features:
- the caching behavior is not suitable for let's say live albums where images are added on an hourly basis. the caching cannot be deactivated & the only way to solve this is to purge the facebook cache manually. Would be nice to include a check from time to time if the cached data is still up to date.

- Facebook albums sometimes have a weird sorting. This can be manually fixed in Facebook (sort most recent), but it would be nice if one could select a sorting in Contao. As far as I could see in the API response, there is also a creation_time or something similar included, for which the array could be sorted.

These are just some suggestions from my side, love your work!

**Comment by fritzmg** [comment]
Yeah, the bug is due to a change in Facebook's API, I will fix that.

> Would be nice to include a check from time to time if the cached data is still up to date.

True, I could add the purging of the cache to TL_CRON, let's say monthly?

Sorting according to a property could be done, yes.

**Comment by reluem** [comment]
ad 1. perfect, thank you!
ad 2. yes would be an option, or maybe even ask the user which interval to use. The particular use case I had today was to implement an easy-to-fill album in almost real-time. I know this is probably not default scenario, but nevertheless changing the cache functionality would be a big plus!
ad 3. okay good to hear!

Thanks for your quick update on this!

**Comment by fritzmg** [comment]
> ad 2. yes would be an option, or maybe even ask the user which interval to use. The particular use case I had today was to implement an easy-to-fill album in almost real-time. I know this is probably not default scenario, but nevertheless changing the cache functionality would be a big plus!

Well, you could set up a daily or minutely cronjob yourself. Just add
```php
$GLOBALS['TL_CRON']['minutely'][] = array('PurgeFacebookCache','purge');
```
to your `initconfig.php`.

**Comment by fritzmg** [comment]
access token problem fixed in c034d9d1524fe5a2f1c6261dda5fcaa781aa0e6e (tagged as `1.3.1`).

**Comment by fritzmg** [comment]
`1.4.0` released:

* added the ability to override caching mechanism for each album
* added the ability to define the sort order of the album
